python/mlcroissant/mlcroissant/_src/core/json_ld.py

```diff
@@ -161,6 +161,8 @@ def recursively_populate_jsonld(entry_node: Json, id_to_node: dict[str, Json]) -
             return recursively_populate_jsonld(entry_node, id_to_node)
         else:
             return entry_node
+    elif isinstance(entry_node, str):
```

Comment:
That wouldn't solve the issue for other non-mapping types.

- Should you consider more 'leaf' types?

```python
elif isinstance(entry_node, (str, float, int, bool)):
  return entry_node
```

- And/or raise an explicit error?

```python
if not isinstance(entry_node, dict):
  raise ValueError(f'Unsupported type for entry_node. Please, open an issue on GitHub to handle: {type(entry_node)}')
```

Reply:
Great point, done.